filebeat/harvester/harvester.go

```diff
@@ -17,12 +17,15 @@ import (
 	"fmt"
 	"regexp"
 
+	"github.com/satori/go.uuid"
+
 	"github.com/elastic/beats/filebeat/config"
 	"github.com/elastic/beats/filebeat/harvester/encoding"
 	"github.com/elastic/beats/filebeat/input"
 )
 
 type Harvester struct {
+	Id                 uuid.UUID
```

Comment:
Yeah, `beat.UUID + harvester id` would be unique for a given instance, __but__ without persistence of the harvester ID, over time (beyond restarts) those IDs would not have meaning.

I think you would want to persist the harvester IDs.